### PR TITLE
Add x-wso2-api-security extension

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
Fix https://github.com/wso2/product-apim/issues/7205

Description:
This PR will introduce an x-wso2-api-security extension to present the application level security types allowed for the API.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java [hunk 1]
```diff
@@ -527,6 +527,7 @@ public String getOASDefinitionForPublisher(API api, String oasDefinition) throws
         if (api.getTransports() != null) {
             swagger.setVendorExtension(APIConstants.X_WSO2_TRANSPORTS, api.getTransports().split(","));
         }
+        swagger.setVendorExtension(APIConstants.SWAGGER_X_WSO2_API_SECURITY, OASParserUtil.getAPISecurity(api.getApiSecurity()));
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java]: can you please check the charactior length is correct?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java [hunk 2]
```diff
@@ -516,6 +516,7 @@ public String getOASDefinitionForPublisher(API api, String oasDefinition) throws
         if (api.getTransports() != null) {
             openAPI.addExtension(APIConstants.X_WSO2_TRANSPORTS, api.getTransports().split(","));
         }
+        openAPI.addExtension(APIConstants.SWAGGER_X_WSO2_API_SECURITY, OASParserUtil.getAPISecurity(api.getApiSecurity()));
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java]: can you please check the charactior length is correct?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OASParserUtil.java [hunk 3]
```diff
@@ -1102,4 +1103,22 @@ public static void removePublisherSpecificInfo(Map<String, Object> extensions) {
...
+    /**
+     * Get Application level security
+     * @param security string
+     * @return List of api security
+     */
+    public static List<String> getAPISecurity(String security){
+        List<String> apiSecurityList = new ArrayList<>();
+        if (security != null) {
+            String[] securityList = security.split(",");
+            for (String s : securityList) {
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OASParserUtil.java]: can we use a meaningful name for s? ex: security?

**Comment by AmaliMatharaarachchi** [comment]
@ruks Thanks. Fixed above review comments in https://github.com/wso2/carbon-apimgt/pull/8143/commits/8136561bef51c04d30b693566710e47b548190a0

**Pull request merged**